src/front/glsl/context.rs

```diff
@@ -1069,35 +1069,138 @@ impl Context {
+
+                // Flush the body of the `false` branch
+                self.emit_flush(&mut reject_body);
+                self.emit_start();
+
+                // We need to do some custom implicit conversions since the two target expressions
+                // are in different bodies
+                if let (
+                    Some((accept_power, accept_width, accept_kind)),
+                    Some((reject_power, reject_width, reject_kind)),
+                ) = (
+                    // Get the components of both branches and calculate the type power
+                    self.expr_scalar_components(parser, accept, accept_meta)?
+                        .and_then(|(kind, width)| Some((type_power(kind, width)?, width, kind))),
+                    self.expr_scalar_components(parser, reject, reject_meta)?
+                        .and_then(|(kind, width)| Some((type_power(kind, width)?, width, kind))),
+                ) {
+                    match accept_power.cmp(&reject_power) {
+                        std::cmp::Ordering::Less => {
+                            self.conversion(&mut accept, accept_meta, reject_kind, reject_width)?;
+                            // The expression belongs to the `true` branch so we need to flush to
+                            // the respective body
+                            self.emit_flush(&mut accept_body);
+                        }
+                        // Technically there's nothing to flush but we need to add some
+                        // expressions that must not be emitted so instead of flushing,
+                        // starting and flushing again, just make sure everything is
+                        // flushed.
+                        std::cmp::Ordering::Equal => self.emit_flush(body),
+                        std::cmp::Ordering::Greater => {
+                            self.conversion(&mut reject, reject_meta, accept_kind, accept_width)?;
+                            // The expression belongs to the `false` branch so we need to flush to
+                            // the respective body
+                            self.emit_flush(&mut reject_body);
+                        }
+                    }
+                }
+
+                // We need to get the type of the resulting expression to create the local,
+                // this must be done after implicit conversions to ensure both branches have
+                // the same type.
+                let ty = parser.resolve_type_handle(self, accept, accept_meta)?;
 
-                self.binary_implicit_conversion(
-                    parser,
-                    &mut accept,
+                // Add the local that will hold the result of our conditional
+                let local = self.locals.append(
+                    LocalVariable {
+                        name: None,
+                        ty,
+                        init: None,
+                    },
+                    Span::default(),
```

Comment:
Would it make sense to use `meta` (i.e. the whole expression) for this `LocalVariable`'s span? I have no idea whether any errors will try to point to it.

Reply:
I specifically was trying to avoid it to prevent the errors to highlight the same span twice but now that I look at it it shouldn't be a problem since the span isn't used in expression analysis